### ISSUE TITLE
Standardize show methods for vcf and chrom objects with cat

### ISSUE THREAD

**Issue by zkamvar** [knausb/vcfR#23]
Feel free to ignore this, for it's mainly a stylistic request, but I've noticed that you use `message()` to display the vcfR object and `print()` to display chromR objects. This results in very different aesthetics in the output, which could lead to confusion from novice users.

```
> chrom
[1] "*****   Class chromR, method Show   *****"
[1] "Name:  Supercontig"
[1] "Length:  1042442"
[1] "Use head(object) for more details."
[1] "*****      End Show (chromR)        *****"
> vcf
***** Object of Class vcfR *****
18 samples
22,031 variants
7.929 percent missing data
*****        *****         *****
```

What's more is that, by using `message()` to to display this information, Rmarkdown understands this as output that's printed line by line in vignettes:

```
## ***** Object of Class vcfR *****
```
```
## 18 samples
```
```
## 22,031 variants
```
```
## 7.929 percent missing data
```
```
## *****        *****         *****
```

My proposed solution would be to use `cat()` for both of them. This has precedence in several other object classes including `DNAbin` (ape), `igraph` (igraph), `tbl_df` (dplyr), and many more.

**Comment by knausb** [comment]
I agree that a single style should adhered to. And I think you can infer from my failure to adhere to this that I've struggled a bit with this issue. Have you seen this:

http://yihui.name/knitr/demo/output/

This is where I've settled on my current opinion that these should be message()s and not cat()s. I'd propose changing to all messages. You bring up one good reason this is suboptimal, but I think the linked site brings up reasons it is better. Opinions?

Thanks!

**Comment by zkamvar** [comment]
I have indeed seen the note about messages and agree to that point that messages to the user should have the option to suppress messages that are side-effects of functions. 

An example in poppr is the behavior of locus_table, where the legend is displayed as a message before the data is returned. 

When using a show method, however, the desired output is a description of the data itself. If the user does not want the output of the data, then they will simply not use the show method. 

What's more is that message is a bit more difficult to control than using cat on a formatted string, making message more effort than it's worth. 

Again, we have different interpretations, but I believe that cat is appropriate for the show method as that is the desired output.

**Comment by zkamvar** [comment]
Just for posterity of the examples I mentioned earlier:

[The default show method in R (when no show method is defined) utilizes cat](https://github.com/wch/r-source/blob/b156e3a711967f58131e23c1b1dc1ea90e2f0c43/src/library/methods/R/show.R#L19-L41)

[`capture.output()` will not capture a message unless explicitly requested](https://stat.ethz.ch/R-manual/R-devel/library/utils/html/capture.output.html) <- I use this in the current version of poppr to help manipulate the show method for snpclone objects.

**Comment by knausb** [comment]
Well Zhian, I knew you liked cats. I guess I didn't realize how much the R people did as well. Fixed in adb25c41c0ed23f083b339063254ee36db6c1b1d.

**Comment by zkamvar** [comment]
:smile_cat:
